main.py

```diff
@@ -306,34 +329,71 @@ def main(cl_arguments):
             # This logic looks strange. We think it works.
             best_epoch = best_epoch[task.name]
             layer_path = os.path.join(args.run_dir, "model_state_eval_best.th")
-            load_model_state(
-                model,
-                layer_path,
-                args.cuda,
-                skip_task_models=task_names_to_avoid_loading,
-                strict=strict)
+            if args.transfer_paradigm == "finetune":
+                # save this fine-tune model with a task specific name
+                finetune_path = os.path.join(args.run_dir, "model_state_%s_best.th" % task.name)
+                os.rename(layer_path, finetune_path)
+
+                # reload the original best model
+                pre_finetune_path = get_best_checkpoint_path(args.run_dir)
+                load_model_state(model, pre_finetune_path, args.cuda, skip_task_models=[], strict=strict)
+            else:
+                load_model_state(model, layer_path, args.cuda,
```

Comment:
Add comment for what's different here?

Reply:
With fine-tuning a model per target task, we need to save the best checkpoint from that target task training to be specific to that target task (else we'll overwrite `model_best_eval_checkpoint` when we start training for the next target task)